### DISCUSSION TITLE
Multi-Language Grammar parser issue

### DISCUSSION THREAD

**Discussion by afidegnum** [tree-sitter/tree-sitter#1499] [Q&A]
I have been reading about tree-sitter, stringtemplate and antlr  but few problems are still not addressed. 

Problem: 

Given a series of tokens i.e Json or typescript, 

```
image": { 
        "src": "Images/Sun.png",
        "name": "sun1",        
    },
```

What provider do I need to generate a code tree with in other languages i.e python, golang, rust etc... 

which will be in a form of the following  python example? 

```
class Image:
  def __init__(self, name, age):
    self.name = name
    self.age = age

```

**Comment by ahelwer** [comment]
It sounds like what you need is a JSON parser then to convert the parsed JSON into a python AST then pretty-print that AST, as is (the latter step) talked about in [this](https://stackoverflow.com/a/7886125/2852699) SO answer.

**Comment by afidegnum** [comment]
Thanks, but the codegen.py specified the is not available, the link is broken. but I understood the concept. What I need is to be able to parse the JSON entry, decode it to AST then can be easily used in various languages besides python. can you suggest another link ?
